### PR TITLE
Eliminate remaining runtime and maintenance debt

### LINKED ISSUE #100: Spec: Eliminate remaining runtime and maintenance debt
## Problem Statement

snstr v0.4 has completed its first cleanup wave, but seven concentrated areas still create avoidable runtime risk and maintenance drag: untested NIP-57 clients, a leaking NIP-86 timeout, noisy and incomplete ephemeral Relay lifecycle handling, shallow security-validator coverage, duplicated package/README command inventories, stale package surfaces, and incompatible diagnostic logger abstractions.

These are individually tractable, but they touch shared Relay and diagnostic seams. They need to land as a compatibility-preserving sequence with focused public-behavior tests.

## Solution

Deliver seven bounded cleanup slices:

1. Expand and migrate to one canonical diagnostic logger without breaking current NIP-02 or NIP-47 consumers.
2. Make the ephemeral Relay quiet by default and harden its close, restart, purge, and socket lifecycle.
3. Consolidate NIP-57 subscription collection and test the public NostrZapClient and ZapClient paths.
4. Exercise security limits through public event and Relay behavior, using direct pure-validator tests only when no public seam exists.
5. Ensure the NIP-86 timeout is cleared on every completion path, including rejected fetches.
6. Collapse duplicated package scripts and README command inventories while retaining useful aliases.
7. Remove the dead NIP-04 shim and the broken redundant NIP-44 JavaScript demo; keep all remaining advertised examples runnable.

## User Stories

1. As a NIP-57 caller, I need every timer and subscription released after success, EOSE, timeout, error, or malformed relay behavior.
2. As a maintainer, I need the public NostrZapClient and ZapClient flows covered rather than their private helpers.
3. As a multi-relay caller, I need all returned subscription IDs cleaned up, not only the first ID.
4. As a caller, I need EOSE and timeout races to settle once without late side effects.
5. As a NIP-86 caller, I need a rejected network request to leave no open timeout handle.
6. As an existing NIP-86 caller, I need current timeout duration and public behavior preserved.
7. As a library consumer, I need importing or using the ephemeral Relay to be quiet unless diagnostics are explicitly injected.
8. As a Relay user, I need repeated close operations to be safe.
9. As a Relay user, I need restart behavior to be predictable after close.
10. As a Relay user, I need purge timers and sockets cleaned up completely.
11. As a security maintainer, I need limits verified through public event and Relay behavior where possible.
12. As a security maintainer, I need direct pure-validator tests only for branches with no meaningful public path.
13. As a security maintainer, I need malformed filters, tags, ranges, keys, and rate-limit boundaries covered.
14. As a contributor, I need one authoritative README command inventory.
15. As a contributor, I need grouped package commands without exact duplicate commands masquerading as separate workflows.
16. As a contributor, I need genuinely useful aliases retained.
17. As a contributor, I need every advertised example to run, or to stop being advertised.
18. As a maintainer, I need unreachable compatibility artifacts removed.
19. As an integrator, I need one canonical diagnostics contract.
20. As an existing NIP-02 integrator, I need my warn-only logger object to continue working.
21. As an existing NIP-47 integrator, I need my structurally compatible logger to continue working.
22. As a web or React Native consumer, I need root export parity for diagnostic types.
23. As a maintainer, I need incompatible diagnostic-contract removal scheduled for the next major release.
24. As a release owner, I need lint, types, tests, coverage, package packing, exports, and examples verified before merge.

## Implementation Decisions

- Preserve public runtime behavior unless this spec explicitly removes a broken or unreachable surface.
- Use a compatibility-first logger migration: expand and migrate now; remove obsolete public logger contracts only in the next major release.
- Treat the core logger as canonical while retaining compatible NIP-specific aliases or narrow structural views during the transition.
- Keep existing NIP-57 and NIP-86 timeout values. Do not add new cancellation, timeout-option, or disposal APIs.
- Consolidate NIP-57 subscription collection behind one internal implementation, clean up every returned subscription ID, and make settlement idempotent.
- Keep the ephemeral Relay public subpath and deepen lifecycle behavior behind its current interface. Remove import-time logging.
- Test security behavior through public events and Relay flows first; directly test pure validators only where no public path can reach the branch. Do not add a global coverage threshold.
- Keep npm, Bun, and pnpm lockfiles and durable run artifacts.
- Remove the dead NIP-04 shim and the broken redundant NIP-44 JavaScript demo.
- Maintain root, web, and React Native export parity, and retain the package export verifier.
- Do not deploy to production.

## Testing Decisions

- NIP-57: test public NostrZapClient and ZapClient methods using observable Relay behavior.
- NIP-86: test the public RelayManagementClient using the existing fetch adapter seam.
- Ephemeral Relay: test public NostrRelay lifecycle and observable Relay behavior.
- Security: use public event and Relay behavior; test direct pure validators only when a public path does not exist.
- Commands and examples: verify through runnable package commands.
- Diagnostics: verify root/web/React Native exports and injected diagnostic behavior.
- Do not test private methods directly.
- Do not introduce a repository-wide coverage threshold.

## Out of Scope

- Removing legacy public logger contracts before the next major version.
- Redesigning NIP-46.
- Splitting the Relay or Nostr classes.
- Adding NIP-57 cancellation or configurable timeout APIs.
- Adding a NIP-86 dispose API.
- Adding a global coverage threshold.
- Changing supported package managers or Node policy.
- Deleting durable agent run records.
- Any production deployment.

## Notes

- Architecture decision: docs/adr/0002-unify-diagnostics-compatibly.md
- A bounded OpenCode GLM-5.2 architecture pass confirmed the proposed seams and compatibility order.
- The requested Grok 4.5 xhigh routes were unavailable in the live local model catalog, so no substitute Grok model was used.

### PR DESCRIPTION
**Pull request by AustinKelsay**
## Summary

Completes cleanup items 1–7 from #100:

- unifies the public diagnostic logger contract without breaking NIP-specific aliases
- makes the ephemeral Relay lifecycle quiet, deterministic, retryable, and cross-runtime
- stabilizes NIP-57 receipt collection across timeout, EOSE, malformed-ID, and cleanup races
- clears NIP-86 request timers on every terminal path
- exercises public security limits at their exact inclusive boundaries
- collapses command/README inventory duplication and verifies the command graph
- removes stale NIP-04/NIP-44 surfaces and repairs maintained example references

Implementation tickets: #101, #102, #103, #104, #105, #106, #107.

## Design and run artifacts

- ADR: `docs/adr/0002-unify-diagnostics-compatibly.md`
- Run ledger: `docs/agents/runs/feature-cleanup-runtime-debt-ledger.md`
- Cross-runtime follow-up: `docs/agents/runs/integrated-bun-compat-session.md`
- Final review: `docs/agents/runs/final-cleanup-review-session.md`

## Verification

- `npm ci`
- command inventory verifier
- ESLint
- root and examples TypeScript checks
- CJS/ESM build and examples build
- package verification: 16 referenced targets, 304 packed files, 49 web modules, 3 guarded Node fallbacks
- live `example:basic` and `example:messaging` groups
- focused final matrix: Jest 103/103 and Bun 103/103
- full Jest with open-handle detection: 72/72 suites, 991/991 tests
- full Bun: 72/72 files, 991/991 tests
- coverage: 72/72 suites, 991/991 tests
  - overall: 76.08% statements, 63.30% branches, 79.50% functions, 76.40% lines
  - `security-validator.ts`: 86.12% statements, 80.00% branches, 84.37% functions, 85.99% lines
- branch diff secret scan and `git diff --check`

## Review

- independent Codex spec/behavior/security review: PASS, zero actionable findings
- independent Codex standards/Fowler review: PASS, zero actionable findings
- OpenCode GLM-5.2 max branch review: PASS, zero actionable findings
- local CodeRabbit: two completed rounds; all nine findings fixed or evidence-rejected
- a third local CodeRabbit confirmation was service-rate-limited, so this PR requests the required hosted full review
- Grok was not substituted because the required Grok 4.5 xhigh routes were absent from the live catalog

## Risks

- `npm ci` retains the existing audit baseline of 13 dependency vulnerabilities (2 low, 3 moderate, 8 high). Dependency locks were intentionally not changed and no potentially breaking audit fix was applied.
- This is a large cleanup diff, so hosted CI and CodeRabbit are required before merge.
- No deployment is part of this PR.

Closes #100

## Summary by CodeRabbit

* **New Features**
  * Added shared diagnostic logger support with `ConsoleLogger`/`DiagnosticLogger` exports across platforms.
  * Introduced warn-only `WarningLogger` (kept as a deprecated alias for compatibility).
  * Enhanced NIP-57 clients with optional diagnostics while staying quiet by default.
  * Improved relay lifecycle shutdown/disconnect coordination and command-reference verification.

* **Bug Fixes**
  * Hardened relay-management request timeout cleanup and zap-receipt collection edge cases.

* **Documentation**
  * Consolidated README command documentation into a single command reference; removed outdated demos and examples.

* **Tests**
  * Expanded multi-runtime (Jest/Bun) and type-compatibility coverage.

### CODE REVIEW

**Comment by AustinKelsay** [comment]
@coderabbit full review

**Comment by AustinKelsay** [comment]
CI follow-up: the first hosted matrix exposed a Node 16-only test portability gap. Node 16 has no built-in `globalThis.fetch`, so three new NIP-57 tests could not spy on it. Commit `039ae8e` installs a scoped mock property only when absent and removes it in teardown; production code is unchanged.

Fresh verification:
- actual Node 16 Jest: NIP-57 client 14/14
- current Node Jest: 14/14 with open-handle detection
- Bun: 14/14
- lint, TypeScript, and `git diff --check`: passed

The previous Node 20 job had already passed all 72 suites / 991 tests before the matrix was canceled by fail-fast from Node 16. The pushed fix has started a fresh hosted matrix.

**Comment by AustinKelsay** [comment]
Second CI portability follow-up: Node 16 lacks `http.Server.closeAllConnections()`, which the new native Relay test helper used as an extra sweep after it had already tracked, destroyed, and awaited every socket. Commit `1dd8428` makes that redundant modern-runtime call conditional.

Fresh verification:
- actual Node 16 Jest with open-handle detection: 6/6
- current Node Jest with open-handle detection: 6/6
- Bun: 6/6
- lint, TypeScript, and `git diff --check`: passed

Node 18's prior rerun completed all 72 suites / 991 tests before matrix fail-fast canceled the job. A fresh hosted matrix is starting from `1dd8428`.

**Comment by AustinKelsay** [comment]
Final gate status:

- Node 16: success
- Node 18: success
- Node 20: success, including coverage, library build, package verification, and examples build
- Bun: success, including library build and package verification
- hosted CodeRabbit full review: finished successfully with zero inline findings
- independent final spec and standards re-reviews of both CI portability fixes: PASS

PR head `1dd8428` is clean and ready for maintainer review into `staging`.

**Comment by AustinKelsay** [comment]
@coderabbitai review

File: tests/nip02/nip02.test.ts [hunk 1]
```diff
@@ -363,6 +363,31 @@ describe("NIP-02: Contact Lists", () => {
       expect(contact4?.relayUrl).toBe("ws://valid.but.different.com");
     });
 
+    it("routes warnings through an unchanged warn-only logger", () => {
+      const warnings: Array<{ message: string; value?: string }> = [];
```
[the remark this answers is not in the document]
Comment (AustinKelsay) [reply tests/nip02/nip02.test.ts:367]: Verified against current HEAD: this is a false positive. There is exactly one `warnings` declaration at tests/nip02/nip02.test.ts:367. Root TypeScript, lint, and the hosted Node 16/18/20 matrices all compile this file successfully, so no code change was made.

File: tests/nip57/client.test.ts [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+        expect(nostr.getRelay(relay.url)?.getSubscriptionIds().size).toBe(0);
+      } finally {
+        nostr.disconnectFromRelays();
+        await relay.close();
+        if (!isBun) {
+          if (hadBun) globals.Bun = previousBun;
+          else delete globals.Bun;
+        }
+      }
+    });
```
[the remark this answers is not in the document]
Comment (AustinKelsay) [reply tests/nip57/client.test.ts:241]: Fixed in 4c38d62. Both tests now enter try/finally before mutating `globalThis.Bun`, starting the Relay, constructing Nostr, or creating the receipt; cleanup conditionally disconnects/closes initialized resources and restores the global. Verified with Jest 14/14 plus open-handle detection, Bun 14/14, lint, TypeScript, and git diff --check.

**Comment by AustinKelsay** [comment]
@coderabbitai review

**Comment by AustinKelsay** [comment]
Three-pass CodeRabbit review complete:

1. `6b27be20-d36d-4805-a844-f95c04ff6fe1` — full 54-file pass; two findings.
   - Valid cleanup-scope finding fixed in `4c38d62`.
   - Duplicate-declaration finding verified false and withdrawn by CodeRabbit.
2. `184fb58c-2c8e-4559-b630-b035b65e458d` — incremental fix review; no new actionable finding, and CodeRabbit accepted both resolutions.
3. `60a1a554-5206-4928-aad2-f28eece8ad26` — final confirmation; “No actionable comments were generated.”

Final CI is green on Node 16, Node 18, Node 20, and Bun. Focused Jest/Bun NIP-57 tests, lint, TypeScript, and diff checks also passed locally.

Grok orchestration was preflighted as requested, but the required `grok-4.5-xhigh` and `grok-4.5-fast-xhigh` routes were absent from the live CLI catalog, so no weaker model was substituted.
